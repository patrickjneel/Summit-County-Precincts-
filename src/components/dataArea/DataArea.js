import React, { Component } from 'react';
import { connect } from 'react-redux';
import './data-area.css';


const DataArea = (props) => {

  console.log(props);
  let foundAddress = props.address.find(address => {
    const street = new RegExp('\\s*'+props.coords.address+'\\s*', 'i');
    const houseRegex = new RegExp(/\d+/);
    const getHouse = props.coords.fullAddress.match(houseRegex);
    let houseNumber;

    if(getHouse) {
      houseNumber = parseInt(getHouse[0]);
    }
   
    if (
      street.test(address.baseAddress) && 
      houseNumber >= address.HouseFrom &&
      houseNumber <= address.houseTo
    ) {
      console.log('found correct house');
      return true;
    }

    return false;
  });

  console.log(foundAddress);
  // console.log(props.coords);
//if found house return a bunch og jsx
//else return same side panel, but return new address

 if(props.coords.address === foundAddress) {
  return <div>{foundAddress}</div>;
 }
  return (
      <div className="data-area">
        <h2>Home Address Precinct Information</h2>
        <table>
          <tbody>
            <tr>
              <td className="key">Address:</td>
              <td className="value">{props.coords.address}</td>
            </tr>
            <tr>
              <td className="key">Precinct:</td>
              <td className="value">{foundAddress.precinct.toString().slice(-2)}</td>
            </tr>
            <tr>
              <td className="key"></td>
              <td className="value">Congressional 2</td>
            </tr><tr>
              <td className="key"></td>
              <td className="value">State Senate 8</td>
            </tr>
            <tr>
              <td className="key"></td>
              <td className="value">State House 61</td>
            </tr>
            <tr>
              <td className="key"></td>
              <td className="value">Judical 5</td>
            </tr>
            <tr>
              <td className="key"></td>
              <td className="value">Dillon</td>
            </tr>
            <tr>
              <td className="key"></td>
              <td className="value">Summit School RE 1</td>
            </tr>
            <tr>
              <td className="key"></td>
              <td className="value">Colorado Mountain College</td>
            </tr> 
            <tr>
              <td className="key"></td>
              <td className="value">Lake Dillon Fire Protection</td>
            </tr>
          </tbody>
        </table>
      </div>
    )
}

export const mapStateToProps = (store) => {
  return {
    address: store.address,
    coords: store.coords
  }
}

export default connect(mapStateToProps, null)(DataArea);
