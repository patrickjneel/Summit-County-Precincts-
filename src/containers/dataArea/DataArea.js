import React, { Component } from 'react';
import { connect } from 'react-redux';
import './data-area.css';
import PropTypes from 'prop-types';


const DataArea = (props) => {
  let foundAddress = props.address.find(address => {
    const street = new RegExp('\\s*'+props.coords.address+'\\s*', 'i');
    const houseRegex = new RegExp(/\d+/);
    const getHouse = props.coords.fullAddress.match(houseRegex);
    let houseNumber;

    console.log(street)
    console.log(getHouse)

    if(getHouse) {
      houseNumber = parseInt(getHouse[0]);
    }
   
    if (
      street.test(address.baseAddress) && 
      houseNumber >= address.HouseFrom &&
      houseNumber <= address.houseTo
    ) {
      return true;
    }

    return false;
  });

let splitInfo;
if(foundAddress) {
  splitInfo = foundAddress.splitData.map(thing => {
    console.log(thing)
    if(thing.splitNumber === foundAddress.splitCode) {
      
      return <div className="split">
              <span className="value">{thing.split[0]}</span> 
              <span className="value">{thing.split[1]}</span> 
              <span className="value">{thing.split[2]}</span> 
              <span className="value">{thing.split[3]}</span> 
              <span className="value">{thing.split[4]}</span> 
              <span className="value">{thing.split[5]}</span> 
              <span className="value">{thing.split[6]}</span> 
              <span className="value">{thing.split[7]}</span> 
            </div>
    }
  })
}
  
    if(!foundAddress) {
      return <div className="data-area">
                Please search for home address
                <img src={require("../../assets/sclogo.png")} />
             </div>
    }

  return (
      <div className="data-area">
        <h2>Home Address Precinct Information</h2>
        <table>
          <tbody>
            <tr>
              <td className="key">Address:</td>
              <td className="value">{props.coords.fullAddress}</td>
            </tr>
            <tr>
              <td className="key">Precinct:</td>
              <td className="value">{foundAddress.precinct.toString().slice(-2)}</td>
            </tr>
          </tbody>
        </table>
        <span>
          {splitInfo}
        </span>
      </div>
    )
}

export const mapStateToProps = (store) => {
  return {
    address: store.address,
    coords: store.coords
  }
}

export default connect(mapStateToProps, null)(DataArea);

DataArea.propTypes = {
  address: PropTypes.array,
  coords: PropTypes.object
};
           
