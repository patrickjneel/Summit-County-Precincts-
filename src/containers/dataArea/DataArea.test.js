import React from 'react';
import { shallow } from 'enzyme';
import { DataArea, mapStateToProps, mapDispatchToProps } from './DataArea';


describe('Data Area Test', () => {
  it.skip('should be defined', () => {
    const mockFunc = jest.fn();
    const renderedData = shallow(<DataArea address={mockFunc}/>);
    expect(renderedData).toBeDefined();
  });

  it.skip('should match the snap shot', () => {
    const mockFunc = jest.fn();
    const renderedData = shallow(<DataArea address={mockFunc}/>);

    expect(renderedData).toMatchSnapshot();
  });
});

describe('Map State To Props Test', () => {
  it('should receive location data from the store', () => {
    const mockStore = {
      coords: {
        address: "Buffalo St",
        lat: 39.05,
        lng: -106.06,
        fullAddress: '200 Buffalo St Dillon, CO'
      }
    };
    const expected = mapStateToProps(mockStore);
    expect(expected.coords).toEqual(mockStore.coords)
  });

  it('should receive address data from the store', () => {
    const mockStore = {
      address: [
      {
        county:"SUMMIT",
        HouseFrom: 701,
        houseTo: 889,
        ruleType: 0,
        fullAddress: [{house: '701 e annemone trl'}],
        precinct: 2086159008,
        splitCode: 26
      }
      ]
    }
    const expected = mapStateToProps(mockStore)
    expect(expected.address).toEqual(mockStore.address)
  });
});
