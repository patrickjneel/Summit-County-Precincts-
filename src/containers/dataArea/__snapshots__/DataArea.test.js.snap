// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Area Test should match the snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    address={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Please search for home address",
        <img
          src="sclogo.png"
/>,
      ],
      "className": "data-area",
    },
    "ref": null,
    "rendered": Array [
      "Please search for home address",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "sclogo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Please search for home address",
          <img
            src="sclogo.png"
/>,
        ],
        "className": "data-area",
      },
      "ref": null,
      "rendered": Array [
        "Please search for home address",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "sclogo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
